aptos-move/framework/aptos-token/sources/property_map.move

```diff
@@ -69,6 +72,42 @@ module aptos_token::property_map {
         simple_map::borrow(&map.map, key)
     }
 
+    public fun read_string(map: &PropertyMap, key: &String): String {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"0x1::string::String"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_string(prop.value)
+    }
+
+    public fun read_u8(map: &PropertyMap, key: &String): u8 {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"u8"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_u8(prop.value)
+    }
+
+    public fun read_u64(map: &PropertyMap, key: &String): u64 {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"u64"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_u64(prop.value)
+    }
+
+    public fun read_address(map: &PropertyMap, key: &String): address {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"address"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_address(prop.value)
+    }
+
+    public fun read_u128(map: &PropertyMap, key: &String): u128 {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"u128"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_u128(prop.value)
+    }
+
+    public fun read_bool(map: &PropertyMap, key: &String): bool {
+        let prop = borrow(map, key);
+        assert!(prop.type == string::utf8(b"bool"), error::invalid_state(ETYPE_NOT_MATCH));
+        from_bcs::to_bool(prop.value)
+    }
+
```

Comment:
This feels like a lot of copypasta, maybe we can minimize this with:

```
public fun read<T>(map: &PropertyMap, key: &String): T {
    let prop = borrow(map, key);
    assert!(prop.type == type_info::typename<T>(), error::invalid_state(ETYPE_NOT_MATCH));
    from_bcs::to_t(prop.value)
}
```

we would need to add 
```
public fun to_t<T>(...)
```
and
```
fun is_allowed<T>(): bool {
    let typename = type_info::typename<T>();
    typename == "bool" ||
        typename == "..."
        ...;
}
```

Reply:
I think this was the original PR I shared. @wrwg  mentioned it would nice to have these explicit types in the function. I am fine with either way. But prefer not going back and forth on the same change

Reply:
no yet. But I heard that people plan to put more metadata all on chain now. want to make the propertyMap more useful.